unlock-app/src/services/unlockProvider.js

```diff
@@ -64,6 +64,11 @@ export default class UnlockProvider extends providers.JsonRpcProvider {
     return sigUtil.personalSign(privateKey, { data })
   }
 
+  eth_signTypedData([_, data]) {
```

Comment:
This was the key missing piece, RPCProvider doesn't seem to implement this method so we needed an internal implementation